async_downloads/static/js/ws_async_downloads.js

```diff
@@ -0,0 +1,185 @@
+$(function () {
+    function readablePeriod(timeStamp) {
+        //determine the timezone offset the browser applies to Date()
+        const offset = new Date().getTimezoneOffset() * 60000;
+
+        const diff = Date.now() - Date.parse(timeStamp) + offset;
+        function suffix(number) {
+            return (number > 1 ? "s" : "") + " ago";
+        }
+        let temp = diff / 1000;
+        const years = Math.floor(temp / 31536000);
+        if (years) return years + " year" + suffix(years);
+        const days = Math.floor((temp %= 31536000) / 86400);
+        if (days) return days + " day" + suffix(days);
+        const hours = Math.floor((temp %= 86400) / 3600);
+        if (hours) return hours + " hour" + suffix(hours);
+        const minutes = Math.floor((temp %= 3600) / 60);
+        if (minutes) return minutes + " minute" + suffix(minutes);
+        const seconds = Math.floor(temp % 60);
+        if (seconds) return seconds + " second" + suffix(seconds);
+        return "now";
+    }
+
+    function calculateTimeStamps() {
+        $(".download-timestamp").each(function () {
+            const timeStamp = $(this).attr("value");
+            $(this).text(readablePeriod(timeStamp));
+        });
+    }
+
+    function initializeAllDownloads(data) {
+        $("#async-downloads .dropdown-menu").empty();
+        data.forEach(function (dl) {
+            initDownload(dl);
+            updateDownload(dl);
+        });
+        if ($("#downloads-dropdown > div").length === 0) {
+            $("#downloads-dropdown").append(
+                '<a class="dropdown-item" href="">No downloads</a>'
+            );
+        } else {
+            calculateTimeStamps();
+        }
+    }
+
+    function initDownload(data) {
+        $("#async-downloads").addClass("in-progress");
+        const dropdown = $("#downloads-dropdown");
+        // remove 'No downloads' msg if exists in dropdown
+        dropdown.find("a").remove();
+        const download = data.download;
+        const download_div = `
+        <div id="download-${data.downloadKey}" class="download-container" 
+            data-url="${download.url}" data-filepath="${download.filepath}" complete="false">
+            <div class="download-content ${download.css_class}">
+                <div class="download">
+                    <span class="download-name">${download.name}</span>
+                    <span class="download-timestamp" value="${download.timestamp}">now</span>
+                </div>
+                <span class="download-clear"><i class="fa fa-times-circle"></i></span>
+
+                <div class="progress" style="height: 4px;">
+                    <div class="progress-bar progress-bar-striped bg-info progress-bar-animated" role="progressbar"
+                        style="width: 0%" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100"></div>
+                </div>
+            </div>
+        </div> `;
```

Comment:
What do you think about sending the HTML in the WebSocket message instead of building it here? That would mean we don't have to repeat the HTML here and can use the existing template.